src/multi-level-checkbox-editor/node-box.tsx

```diff
@@ -79,29 +86,58 @@ const NodeBox: React.FunctionComponent<Props> = (props) => (
-                        }`}
-                    >
-                        <div>
-                            <TriStateCheckbox
-                                id={`node-${node.id}`}
-                                value={getNodeState(node)}
-                                onChange={props.setNodeValue.bind(null, node)}
-                            />
-                            {node.label}
-                        </div>
-                        {hasChildren(node) ? (
-                            <span className="child-node-count" data-qa={`node-label-${node.id}`}>{`${countCheckedNodes(
-                                node
-                            )} / ${countAllNodes(node)}`}</span>
-                        ) : null}
-                    </li>
+                    <div>
+                        <li
+                            key={node.id}
+                            role="listitem"
+                            data-qa={`node-${node.label}`}
+                            onClick={props.onNodeClick.bind(null, node)}
+                            className={`node-list-item ${node.isHighlighted ? 'highlighted' : ''} ${
+                                hasChildren(node) ? '' : 'no-child-node'
+                            }`}
+                        >
+                            <div>
+                                <TriStateCheckbox
+                                    id={`node-${node.id}`}
+                                    value={getNodeState(node)}
+                                    onChange={props.setNodeValue.bind(null, node)}
+                                />
+                                <NodeLabel label={node.label} labelInformation={node.labelInformation} />
+                            </div>
+                            {hasChildren(node) ? (
+                                <span
+                                    className="child-node-count"
+                                    data-qa={`node-label-${node.id}`}
+                                >{`${countCheckedNodes(node)} / ${countAllNodes(node)}`}</span>
+                            ) : null}
+                        </li>
+                        {node.labelInformation && node.labelInformation.labels.length > 1 && (
+                            <div className={'node-tooltip-note'}>
+                                <div className={'note-info'}>
+                                    <div className={'label-information-note'}>
+                                        <div>{node.labelInformation.note}</div>
+                                    </div>
+                                </div>
+
+                                <div className={'node-tooltip'}>
+                                    <Tooltip
+                                        placement="left"
+                                        text={
+                                            <div className={'node-tooltip-container-box'}>
```

Comment:
It looks nice, just one miner thing I recommend creating a separate component and use it here, just for better readability.

Reply:
ok sure